da/blob/factory.go

```diff
@@ -81,22 +87,46 @@ func (f *SidecarFactory[BeaconBlockT, _]) BuildSidecars(
 		sidecars    = make([]*types.BlobSidecar, numBlobs)
 		body        = blk.GetBody()
 		g           = errgroup.Group{}
+		//nolint:errcheck // should be safe
+		header = any(blk.GetHeader()).(*constypes.BeaconBlockHeader)
 	)
 
 	startTime := time.Now()
 	defer f.metrics.measureBuildSidecarsDuration(
 		startTime, math.U64(numBlobs),
 	)
+
+	// Contrary to the spec, we do not need to sign the full
+	// block, because the header embeds the body's hash tree root
+	// already. We just need a bond between the block signer (already
+	// tied to CometBFT's ProposerAddress) and the sidecars.
```

Comment:
q: can you clarify why we deviate from the specs? Granted this is safe, is there a reason why we should not be fully compliant here?

Reply:
Spec-compliant implementations have the beacon block signed from the beginning, therein the beacon block factory always provide a signed envelope. In Berachain, we have never carried out full block signatures from within the app because the signature is done by CometBFT on a CometBFT block, not on the Beacon block (the CometBFT payload). Therefore, it is just a matter of opportunism to sign only the bare minimum (the block header) for the requirement.
Of course, we could sign the full block, given that the HTR of the body is already there in the header (and possible benefit from a tighter bond between the block and the proposer), but at this point the CometBFT signature becomes redundant (and a bloat).